manticore/manticore.py

```diff
@@ -168,10 +168,13 @@ def __init__(self, path_or_state, argv=None, workspace_url=None, policy='random'
         self.forward_events_from(self._executor)
 
         if isinstance(path_or_state, str):
-            assert os.path.isfile(path_or_state)
+            if not os.path.isfile(path_or_state):
+                raise Exception("The file %s doesn't exist" % path_or_state)
             self._initial_state = make_initial_state(path_or_state, argv=argv, **kwargs)
         elif isinstance(path_or_state, State):
             self._initial_state = path_or_state
+        else:
+            raise TypeError('path_or_state must be either str or State while it is %s' % type(path_or_state))
```

Comment:
cool, I like the idea of being more explicit here, rather than relying on the below TypeError which is a slightly more indirect error message. 

for phrasing, this is SO nitpicky, but can we pls match the phrasing used by the `Manticore.add_hook` typecheck? 
https://github.com/trailofbits/manticore/blob/master/manticore/manticore.py#L457
"path_or_state must be either a str or State, not {}".format(type(path_or_state).__name__)
(also note how I did the `__name__` at the end, which should be a little nicer output)
it's very nitpicky, but i think attention to detail like this will be good as we strive to make manticore a more mature project :)